syntax = "proto3";

package learner;

service Learner {
  rpc Learn (LearnRequest) returns (LearnReply) {}
  rpc CreateWordCloud (CreateWordCloudRequest) returns (CreateWordCloudReply) {}
}

// The request message containing the user's hatenaID.
message LearnRequest {
  string hatena_id = 1;
}

message LearnReply {
  bool learned = 1;
}

// The request message containing the user's hatenaID.
message CreateWordCloudRequest {
  string hatena_id = 1;
}

// The response message containing the wordcloud images
message CreateWordCloudReply {
  bytes wordcloud = 1;
}
